src/App.js

```diff
@@ -1,69 +1,63 @@
-import React, { Component } from 'react';
-import { connect } from 'react-redux';
-import { View } from 'react-native';
-import firebase from 'firebase';
-import { Actions } from 'react-native-router-flux';
-import Router from './config/Router';
-import { appVersion } from '../package.json';
+import 'react-native-gesture-handler';
+import React, { useEffect, useRef } from 'react';
+import { useDispatch, useSelector, useStore } from 'react-redux';
+import Router from './router/Router';
+import { View, LogBox } from 'react-native';
 import { Alert } from './components';
-import { apiKey, authDomain, databaseURL, projectId, storageBucket, messagingSenderId, appId } from '@env';
-import { loadUser, getAllMemberAccounts, getEvents, getCommittees, getAllMemberPoints, updateElection } from './ducks';
-
-console.ignoredYellowBox = ['Setting a timer'];
-
-class App extends Component {
-	componentDidMount() {
-		const config = { apiKey, authDomain, databaseURL, projectId, storageBucket, messagingSenderId, appId };
-
-		if (!firebase.apps.length)
-			firebase.initializeApp(config);
-
-		this.verifyLogIn();
-	}
-
-	verifyLogIn() {
-		const { getCommittees, getAllMemberAccounts, getEvents, loadUser, getAllMemberPoints, updateElection } = this.props;
-
-		firebase.auth().onAuthStateChanged(user => {
-			firebase.database().ref('/version').once('value', snapshot => {
-				let correctVersion = snapshot.val() === appVersion;
-
-				if (correctVersion && user) {
-					Actions.main();
-					loadUser();
-					getEvents();
-					getCommittees();
-					getAllMemberAccounts();
-					updateElection();
-					getAllMemberPoints();
-				}
-				else {
-					Actions.login();
-					if (!correctVersion) Alert.alert('Please update your app');
-					if (user) firebase.auth().signOut();
-				}
-				// Actions.splash({ correctVersion, user, signOut: () => firebase.auth.signOut() });
-			});
-		});
-	}
-
-	render() {
-		return (
-			<View style = {{ flex: 1 }}>
-				<Router />
-				<Alert.AlertBox />
-			</View>
-		);
-	}
-}
-
-const mapDispatchToProps = {
+import {
+	initializeFirebase,
+	logoutUser,
+	verifyAppVersion,
+	userStatus,
 	loadUser,
-	getAllMemberPoints,
+	getAllMemberAccounts,
 	getEvents,
 	getCommittees,
-	getAllMemberAccounts,
+	getAllMemberPoints,
 	updateElection
-};
-
-export default connect(() => ({}), mapDispatchToProps)(App);
\ No newline at end of file
+} from './ducks';
+
+LogBox.ignoreAllLogs(true);
+
+export default () => {
+	const { isLoggedIn, hasCorrectVersion } = useSelector(state => state.app);
```

Comment:
whats the use of this line if we're using useStore?

Reply:
To cause the `useEffect` to behave as `componentDidUpdate` when `isLoggedIn` changes. These values are also the ones that are passed in to the router on line 59.